src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -202,7 +202,7 @@ private bool IsServerValidForCluster(ClusterType clusterType, ClusterConnectionM
                         case ClusterConnectionMode.Automatic:
                             if (serverType == ServerType.Standalone)
                             {
-                                return Settings.Scheme == ConnectionStringScheme.MongoDBPlusSrv; // Standalone is only valid in MultiServerCluster when using MongoDBPlusSrv scheme
+                                return _servers.Count == 1 || Settings.Scheme == ConnectionStringScheme.MongoDBPlusSrv;
```

Comment:
Which of the new tests fail if you remove this change?

Reply:
This change is required to pass old and new tests:
* Old:
1. `single/too_old` : https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Core.Tests/Specifications/server-discovery-and-monitoring/tests/single/too_old.json#L2
2. `single/too_new` : https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Core.Tests/Specifications/server-discovery-and-monitoring/tests/single/too_new.json#L2
3. `single/not_ok`: https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Core.Tests/Specifications/server-discovery-and-monitoring/tests/single/not_ok_response.json#L2
4. `single/ls_timeout_Standalone`: https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Core.Tests/Specifications/server-discovery-and-monitoring/tests/single/ls_timeout_standalone.json#L2
5. `single/compatible`: https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Core.Tests/Specifications/server-discovery-and-monitoring/tests/single/compatible.json#L2
* New:
1. `single/discover_Standalone`: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/134/files#diff-ebc2080a5a04dee58bf3254037618628R2

Reply:
The reason why the old above tests fail is here: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/134/files#diff-81e2452642fdc0ce280cdd663e9f6cb4R56

Reply:
But without this change, we have 8 more failed tests (they are new) :)

Reply:
Just to note: I'm not sure which spec requirement is the reason for this change.

Reply:
oh, because we used to force Automatic with a single server to SingleServerCluster.